### ISSUE TITLE
Other (Strata lines, sealion rookeries, etc.)

### ISSUE THREAD

**Issue by nwraring** [afsc-gap-products/Survey-Planning#10]
Instructions for creating chart layers are [here](https://docs.google.com/document/d/1Dyt0soCwV0SBIY6bk4szMGlNDwGNGxQL/edit?usp=sharing&ouid=111519802483611531993&rtpof=true&sd=true).
Instructions for turning shapefiles into Globe files are [here](https://docs.google.com/document/d/11bBlF6z38_F_t2ZXlPxizyUfGrQG2856TLRKfKQQP1w/edit?usp=sharing). 

(Edited by Megsie to include more info, mostly for my own use!)

**Comment by nwraring** [comment]
When will this be added?

**Comment by coleary-noaa** [comment]
The following Lines files were added to both Globe computers and G:\ALEUTIAN\AI 2022\Files for Boats\Globe\Lines:
-globe_paths_2022.mdb
-New_rookery_lines_2021.mdb
-survey_grid_2022.mdb
-Survey2018SSLhapc areas.mdb
-SurveyTowPaths1980through2016.mdb

The following Marks files were added to both Globe computers and G:\ALEUTIAN\AI 2022\Files for Boats\Globe\Marks:
-AI_stations_2022.mdb
-globe_starts_2022.mdb
-NoTransitZones.mdb
-SSl_3nm_Buffer_2021.mdb

Remaining to do:
- add crab pot locations (still waiting on these from Mark Z)
- potentially buoy locations &/or Special Projects zones.
- add instructions to fieldwork folder, temporarily [here](https://docs.google.com/document/d/1Dyt0soCwV0SBIY6bk4szMGlNDwGNGxQL/edit?usp=sharing&ouid=102897708184166605880&rtpof=true&sd=true)
 - return Globe computers to Gary by 01 April

**Comment by MargaretSiple-NOAA** [comment]
Christina sent special project zones on 3/3/22 for stone corals. Other special projects may have to come up with us on a USB drive. I am turning the stone corals csv into a accdb file for globe.

**Comment by MargaretSiple-NOAA** [comment]
Stone corals special project areas added to Globe PC 990 and G Drive:
-special_projects_2022_stone.mdb

Remaining to do:

- add crab pot locations (still waiting on these from Mark Z)
- potentially buoy locations and non-coral Special Projects zones
- add instructions to fieldwork folder, temporarily [here](https://docs.google.com/document/d/1Dyt0soCwV0SBIY6bk4szMGlNDwGNGxQL/edit?usp=sharing&ouid=102897708184166605880&rtpof=true&sd=true)
- return Globe computers to Gary by 01 April

**Comment by MargaretSiple-NOAA** [comment]
No other maps for special projects; remaining to do:

- [x] add crab pot locations (still waiting on these from Mark Z)
- [x] potentially buoy locations
- [x] add instructions to fieldwork folder, temporarily [here](https://docs.google.com/document/d/1Dyt0soCwV0SBIY6bk4szMGlNDwGNGxQL/edit?usp=sharing&ouid=102897708184166605880&rtpof=true&sd=true)
- [x] return Globe computers to Gary by 01 April

**Comment by Ned-Laman-NOAA** [comment]
Took a quick look at the suite of functions I helped to write under the general function tow.path.oracle.globe() and can say that the make.globe.lines() function contains the code for writing Mark's shapefiles (around line 100 and again around line 375). Granted this is all probably deprecated rgdal or some other ancient package code, but I think it may still work...? I would also note in case it isn't clear elsewhere that pretty much all of the Globe layers you make need to be duplicated as shapefiles for the ArcGIS generated charts. If there's anything I can do to help, let me know.

**Comment by MargaretSiple-NOAA** [comment]
Hi @Ned-Laman-NOAA -- per Mark, he has all the Globe-related products he needs for this year, so there is are no more ArcGIS shapefiles that need to be generated for this year. 

The code we ran did generate `towpaths.shp`, `arcstarts.shp`, and `midpoints.shp` automatically, but I thought maybe there was a `.shp` file that had to do with the survey grid. 

Anyway, Mark has what he needs. He sent me the following protected species files that are available. Can you tell me which of these need to go into Globe? **NOTE: EDITED TO BE A BULLET LIST OF WHAT NEEDS TO BE DONE**:

*- [x] CIB_critical_habitat.shp (Cook Inlet beluga whales critical habitat, dated Feb. 14, 2017)*

- NPRWCH.shp (North Pacific Right Whales critical habitat, dated June 16, 2016)
- SeaOtterFinalCH_Project.shp (Sea otter critical habitat, dated June 16, 2016)
- Crit_Hab_Coded_WDPA.shp (western SSLs, dated Oct. 22, 2010)
- New_rookery_lines.shp (dated May 21, 2018; and SSL_3nm_Buffer.shp, dated April 5, 2019). Replace with file from Wayne 2020 (.zip emailed to Cecilia and Ned 3/20/22)

**Comment by Ned-Laman-NOAA** [comment]
Great to hear that Mark has what he needs. That is a key step taking care of.
I think in the past, I have been taking components from the digital chart products process and then building the grid shapefile in ArcGIS. I do believe, however, that @zoyafuso-NOAA may have generated the survey grid shapefile from R in which case I suggest stealing his code and integrating it into these digital chart products! Thanks in advance, Zack.
I'll check off the files it would be nice to have in Globe this year.

**Comment by zoyafuso-NOAA** [comment]
The code to build the shapefile of the AI grids with trawlability information lives [here](https://github.com/zoyafuso-NOAA/AI_planning_stations/blob/main/code/trawlable_grids.R)

**Comment by MargaretSiple-NOAA** [comment]
Thanks Zack. I believe we have a duplicate function in the old Globe R code files so I added a [note](https://github.com/coleary-noaa/Globe/issues/1) to the Globe repo that we could use yours instead (it's a little more polished). 

And thanks Ned for the check boxes!

**Comment by MargaretSiple-NOAA** [comment]
Question for @Ned-Laman-NOAA about SSL rookery files:
We've got two SSL rookery map datasets, one that Mark got from Wayne in 2020, and one that we have from 2021. 
The file that Mark got from Wayne for AI 2020 a few additional SSL rookeries. Rookeries at Semisopochnoi, and two rookeries at the eastern end of Amchitka. 

The file that we copied from GOA 2021 (what is currently on the G Drive) does not have these colonies.

I am tempted to say we should use Mark's file instead, but wanted to ask you which one we should be using. If the Semisopochnoi and Amchtika colonies were abandoned, we don't need to include them, but if we're not sure, maybe we should include them to be more risk-averse? Sorry to be responsible for yet another automated email in your inbox, but c'est la survey!

**Comment by Ned-Laman-NOAA** [comment]
Thanks for letting me know, Megsie.
I agree with your thought about being risk averse and including those rookeries at Semispochnoi and Amchitka.
Can you share Mark's files with me as well - point me to them on the network? I would like to update my layers here for the protected species consultation step in the Scientific Research Permit process.

**Comment by MargaretSiple-NOAA** [comment]
@Ned-Laman-NOAA Mark didn't share his file before he left for the conference but I sent him an email and Bethany may have it too. I will send to you as soon as I have it! Sorry, should have gotten it from him earlier.

**Comment by coleary-noaa** [comment]
Updating remaining to do before survey to bottom of thread for my brain:

Special Habitats bonus
- [x] NPRWCH.shp (North Pacific Right Whales critical habitat, dated June 16, 2016)
- [x] SeaOtterFinalCH_Project.shp (Sea otter critical habitat, dated June 16, 2016)
- [x] Crit_Hab_Coded_WDPA.shp (western SSLs, dated Oct. 22, 2010) - **did not add**; see comment below for why; Bethany will keep layer in ArcMap electronic charts for reference for interested parties
- [x] New_rookery_lines.shp (dated May 21, 2018; and SSL_3nm_Buffer.shp, dated April 5, 2019). Replace with file from Wayne 2020 (.zip emailed to Cecilia and Ned 3/20/22)

Essential
- [x] add crab pot locations (still waiting on these from Mark Z) - **likely carry on**
- [x] add instructions to fieldwork folder, temporarily [here](https://docs.google.com/document/d/1Dyt0soCwV0SBIY6bk4szMGlNDwGNGxQL/edit?usp=sharing&ouid=102897708184166605880&rtpof=true&sd=true)
- [x] return Globe computers to Gary by 01 April

To do in code post-product completion:
- [ ] investigate deprecated functions
- [x] update readme to include shapefile instructions function as well
- [x] merge globe code from Cecilia's repo to GAP group repo

**Comment by coleary-noaa** [comment]
updated rookery 3nm buffer zones & lines for SSL with new union shapefile. added to Files for boats on G drive in 'Marks' folder, named 'SSl_3nm_Buffer_2022.mdb' and 'SSL_rookeries_lines_2022.mdb'

updated rookery 3nm buffer zones & rookery lines for SSL, stone corals special project areas & buoys added to Globe PC 715.

**Comment by coleary-noaa** [comment]
Instructions still in progress but moved to group shared folder [here](https://drive.google.com/drive/folders/1mHTS-GVlmnEGUPBdXCs4WiajsOEO_xAt?usp=sharing)

**Comment by coleary-noaa** [comment]
@Ned-Laman-NOAA not planning to add SSL critical habitat because their critical habitat overwhelms the entire AI habitat; see image below (leaving in the SSL 3nm buffers, SSL rookery lines, Survey2018SSLhapc areas.mdb and NoTranzitZones.mdb in Files for Boats & on Globe computer)


![image](https://user-images.githubusercontent.com/64496320/159373154-e3d6b9ce-7100-45c4-8110-9a672fb60be0.png)

**Comment by Ned-Laman-NOAA** [comment]
Makes sense. I think we can have the layer in the folders and available if needed, but no need to display it.
Basically, more than 3/4 of our stations are in SSL Critical zone and no-one seems to blink; not until we are approaching those 3nm no transit buffers.

**Comment by coleary-noaa** [comment]
ok great; yes the 3 nm buffers and the rookery lines will remain and are updated!

**Comment by coleary-noaa** [comment]
NPRWCH.shp converted to Globe file, added to both to Globe PC 715 and G drive files for boats

**Comment by Ned-Laman-NOAA** [comment]
FYI - Just heard from skipper of the Alaska Provider that they are running the newest version of Globe on the bridge (4.1.5). I wonder if we can get a trial copy of that version to see if our layers display in that version? Can one of you check the ECC website to see if there are trial copies available? There used to be back when.

**Comment by coleary-noaa** [comment]
@Ned-Laman-NOAA After quick webpage browse, seems like there might be demos? http://www.globenav.com/downloads.html

**Comment by Ned-Laman-NOAA** [comment]
Sounds right. Not sure how much functionality the trial/demo versions have, but it's one thing to try.
We could also try talking to the people at ECC. They have typically been very helpful. Maybe we could share a lines or marks file and ask them if it displays in 4.1.5?

**Comment by Ned-Laman-NOAA** [comment]
Chatted with Megsie about this today in a different meeting.
We may also be able to ask ECC Globe for support directly by seeing if anyone there would be willing to try and display one of our marks/lines files in the newer version of Globe just to see if it would show up.
McMurrin was thinking about new copies of Globe, but I don't know if that was in a purchasing sense or not. I will try to remember to ask when I see him tomorrow.

**Comment by coleary-noaa** [comment]
All files that can be completed are completed! Loaded on G drive in Files for Boats Globe folder, as well as on Globe PC 715 and Globe PC 990. Complete list of files included below:</br>
**Lines**
![image](https://user-images.githubusercontent.com/64496320/159585867-be706b64-6c67-495d-8765-7486e27a945d.png)

**Marks**
![image](https://user-images.githubusercontent.com/64496320/159588506-b76446ac-e280-42e6-8ce2-ac54e497dc67.png)

</br>Both Globe computers will be returned to Gary tomorrow (23 March 2022). The crab pot locations file will be created upon receival of info and brought up via thumb drive on Leg 1!

**warning** the sea otter critical habitat layer (Otters_2022.mdb) in the Marks folder is HUGE and makes globe unhappy, and so FPCs shouldn't open it unless really really needed.

**Comment by MargaretSiple-NOAA** [comment]
Re: the newer version of Globe on some of the ships, I talked to Daniel from ECC on the phone this afternoon and he says everything should be compatible with the newer Globe version. I sent him a test file (just some buoys) to test it out on Globe 4.1.5 and will report back here with the result. He says we can also call him when we're at sea.

Cecilia and I will mark our Globe computers tomorrow on campus and return to Gary.

**Comment by Ned-Laman-NOAA** [comment]
Yay! Super that you connected with Daniel. He's been there forever and is very helpful.
Thanks to you both for doing a great job with a difficult project!

**Comment by MargaretSiple-NOAA** [comment]
Daniel says our .mdb files work in Globe 4.1.5 so we're good on that front.

**Comment by Ned-Laman-NOAA** [comment]
Outstanding!!

**Comment by Ned-Laman-NOAA** [comment]
Can you share any habitat areas of particular concern (HAPC) shapefiles you have to the folder here G:\AI-GOA\shapefiles? I need them for the Scientific Research Permit (SRP) consultation that will begin once we have a named second vessel.
I see from your labels for the mdb versions that you've labeled these SSLHAPC, but my understanding is that these HAPCs originated from areas of high sponge/coral concentration and the SSL zones are something different. Thoughts?

**Comment by Ned-Laman-NOAA** [comment]
Hi all. I got these from Mark Z and they now live @G:\AI-GOA\shapefiles\AI_Coral_HPA

**Comment by MargaretSiple-NOAA** [comment]
Hi Ned -- sorry I dropped the ball on these; it was on my to do list to get them to you and other stuff piled up! Glad you got what you needed from Mark.

**Comment by MargaretSiple-NOAA** [comment]
Sounds from @bethanyriggle (from, in turn, Pam Goddard) like we may not have the most updated sea lion rookery shapefiles and we may need to change them. Bethany is going to check "updated" files from Pam in GIS and then we may have to reformat/reimport those if there are big differences between the two files. I will be in the office tomorrow (tuesday) and will troubleshoot w Bethany.

**Comment by bethanyriggle** [comment]
From Pam Goddard, for all up-to-date SSL Rookeries and Haulouts, use these links: 
https://www.fisheries.noaa.gov/alaska/marine-mammal-protection/steller-sea-lion-data
https://www.fisheries.noaa.gov/inport/item/17921

(For use with future surveys)

**Comment by Ned-Laman-NOAA** [comment]
I seem to recall Rebecca Reuter directing me to a website with shapefiles for the rookeries and they were based on that 2015 pub. However, I'm not finding it in the mountain of old email so one of you might contact Rebecca for her take on the most up to date rookery versions. If I find that link I'll send it your way.

**Comment by MargaretSiple-NOAA** [comment]
Adding checklist for crab pots (need to prep these for Globe) -- @MarkZimmermann-NOAA feel free to edit this list as needed):

- [x] Alaska Trojan
- [x] Early Dawn West (called "wag storage april 22.mdb")
- [x] Early Dawn East
- [x] Pat Lee's Early Dawn storage
- [x] Pat Lee's Erla N storage

**Comment by MargaretSiple-NOAA** [comment]
All currently avail crab pot files have been formatted for Globe; I will test them on Globe computer on 4/29 and if good, will add to G Drive files for boats and Globe computers.

**Comment by Ned-Laman-NOAA** [comment]
Sweet!

**Comment by coleary-noaa** [comment]
Crab pot location details for reference/comparison to Globe:
key: 
![image](https://user-images.githubusercontent.com/64496320/166065664-791ee880-2e69-430e-bd15-15d83e9f8acf.png)

Overview: 
![image](https://user-images.githubusercontent.com/64496320/166065727-cabc4df3-4c7f-42fe-a612-62de06132c2f.png)

PL_ED & PatLees_gear
![image](https://user-images.githubusercontent.com/64496320/166065796-b0e47888-2d42-4cae-abb3-d36571e08b83.png)

AK Trojan:
![image](https://user-images.githubusercontent.com/64496320/166065633-0c4f2f5a-8027-4358-8dfb-af808c5297eb.png)

**Comment by Ned-Laman-NOAA** [comment]
That there is a lot of crab pots! Anybody overlay this with station allocation yet to see how many conflicts we are looking at?

**Comment by MarkZimmermann-NOAA** [comment]
Yes, these all look good!

**Comment by MargaretSiple-NOAA** [comment]
They look good in GIS... but we have to get them to look good in Globe (I'm working on it!). @Ned-Laman-NOAA it **is** a lot of crab pots and I think Mark has been looking at overlap!

**Comment by MargaretSiple-NOAA** [comment]
It took a minute but I figured out why some of those crab pot lines were missing-- it was an issue with the R code I was using to produce the csv files. Now it is [fixed ](https://github.com/afsc-gap-products/Globe/commit/2a40fac2e5ed2fcd648ad27560e7a599479b54b0)and looks like it should! NB, for Cecilia, I couldn't really figure out what was happening, but after a certain number of rows it was producing replicates of the start, mid, and end lat/longs for lines. The janky but new code in [crabpots/convert_crabpot_lines.R](https://github.com/afsc-gap-products/Globe/blob/main/crabpots/convert_crabpot_lines.R) was used to produce the final .mdb files. I would loooooove if someone is willing to check my final products in addition to me. I will send you the .mdx file if needed!

## Files
All these .mdb files are on the G Drive (G:\ALEUTIAN\AI 2022\Files for Boats\Globe\Lines\CrabLines) and on Globe computer 990 (the one I took home earlier this season - it is currently in my office on campus). I don't know if I will be in to the office until later next week, so if we need to get the Globe computers loaded up quickly, someone else might want to grab the G Drive files and put them on the other Globe computer.

## Screenshots to check
AK Trojan:
![ak-trojan-check2](https://user-images.githubusercontent.com/73858992/166079814-12fc6e1d-8dfd-4643-b723-78e58bd8962a.PNG)

Early Dawn West:
![wag-storage-check](https://user-images.githubusercontent.com/73858992/166082573-1d27e4c8-cda5-48e0-91c2-41abaf2486d6.PNG)

Early Dawn East:
![EarlyDawnEast_check](https://user-images.githubusercontent.com/73858992/166082806-bbd268c6-2c4b-4cda-b44a-696e59da5e2d.PNG)

Pat Lee's Early Dawn:
![PL_ED_check](https://user-images.githubusercontent.com/73858992/166082862-a2479b0c-7e22-4712-b356-41d8c09d7337.PNG)

Pat Lee's Erla N:
![PL_ErlaN_check](https://user-images.githubusercontent.com/73858992/166082897-3329d57b-4a1f-412d-926e-41eca113ebd3.PNG)

## Shapefile map
Full GIS map with all the shapefiles I have so far:
![CrabPots](https://user-images.githubusercontent.com/73858992/166083219-7a9c5f57-f7db-4592-abb8-52d5f442697f.png)

**Comment by Ned-Laman-NOAA** [comment]
Great detective work, Megsie!
I'll be in on Tuesday next week and can move files around if needed. Where is the other Globe machine?

**Comment by coleary-noaa** [comment]
Gary should have both Globe machines for AI!

**Comment by coleary-noaa** [comment]
> It took a minute but I figured out why some of those crab pot lines were missing-- it was an issue with the R code I was using to produce the csv files. Now it is [fixed ](https://github.com/afsc-gap-products/Globe/commit/2a40fac2e5ed2fcd648ad27560e7a599479b54b0)and looks like it should! NB, for Cecilia, I couldn't really figure out what was happening, but after a certain number of rows it was producing replicates of the start, mid, and end lat/longs for lines. The janky but new code in [crabpots/convert_crabpot_lines.R](https://github.com/afsc-gap-products/Globe/blob/main/crabpots/convert_crabpot_lines.R) was used to produce the final .mdb files. I would loooooove if someone is willing to check my final products in addition to me. I will send you the .mdx file if needed!
> 
> ## Files
> All these .mdb files are on the G Drive (G:\ALEUTIAN\AI 2022\Files for Boats\Globe\Lines\CrabLines) and on Globe computer 990 (the one I took home earlier this season - it is currently in my office on campus). I don't know if I will be in to the office until later next week, so if we need to get the Globe computers loaded up quickly, someone else might want to grab the G Drive files and put them on the other Globe computer.
> 
> ## Screenshots to check
> AK Trojan: ![ak-trojan-check2](https://user-images.githubusercontent.com/73858992/166079814-12fc6e1d-8dfd-4643-b723-78e58bd8962a.PNG)
> 
> Early Dawn West: ![wag-storage-check](https://user-images.githubusercontent.com/73858992/166082573-1d27e4c8-cda5-48e0-91c2-41abaf2486d6.PNG)
> 
> Early Dawn East: ![EarlyDawnEast_check](https://user-images.githubusercontent.com/73858992/166082806-bbd268c6-2c4b-4cda-b44a-696e59da5e2d.PNG)
> 
> Pat Lee's Early Dawn: ![PL_ED_check](https://user-images.githubusercontent.com/73858992/166082862-a2479b0c-7e22-4712-b356-41d8c09d7337.PNG)
> 
> Pat Lee's Erla N: ![PL_ErlaN_check](https://user-images.githubusercontent.com/73858992/166082897-3329d57b-4a1f-412d-926e-41eca113ebd3.PNG)
> 
> ## Shapefile map
> Full GIS map with all the shapefiles I have so far: ![CrabPots](https://user-images.githubusercontent.com/73858992/166083219-7a9c5f57-f7db-4592-abb8-52d5f442697f.png)

I'll see if I can take a look and figure out why replicates were happening! Let me know what .mdx file is needed and I can check today or tomorrow. Awesome job sleuthing that out!

**Comment by MargaretSiple-NOAA** [comment]
Thanks Cecilia. I emailed you the .mdx file with all the crabpots included, which should make it easier to check. Originally, I had adapted the crabpot line code from `SSLRookeries/format_rookery_lines.R`. The part that I changed is the creation of the table for the .csv, which is now in [this tidyverse variation](https://github.com/afsc-gap-products/Globe/blob/c9c5b0e86690c31705fe81aebef1dc2adfefaccd/crabpots/convert_crabpot_lines.R#L9-L30)-- not the most graceful but it was the only way I could rebuild it in a way I understood!

**Comment by MargaretSiple-NOAA** [comment]
> Great detective work, Megsie! I'll be in on Tuesday next week and can move files around if needed. Where is the other Globe machine?

Yes-- Gary has one and the other is in my office (1172).

**Comment by coleary-noaa** [comment]
@MargaretSiple-NOAA double checked .mxd file relative to globe images and all crab pot lines match, and R conversion code looks good to go!

**Comment by MargaretSiple-NOAA** [comment]
OK the updated rookery lines are now in Files for Boats and in our Globe repo. Cecilia has gotten the last two crab boat files and put them on FFB too! I am hereby marking this task as done 🚀

**Comment by MarkZimmermann-NOAA** [comment]
Yes, we have several station conflicts.
I have been checking them out.

I think it would be a good exercise for all of us involved in the mapping
to go through pot/station/trawl overlaps.

Still waiting on files from the last two boats (Aleutian #1 and Patricia
Lee) who will be fishing for another month or so.
-Mark
